fn_alienvault_otx/fn_alienvault_otx/util/alienvault_modules.py

```diff
@@ -0,0 +1,153 @@
+# (c) Copyright IBM Corp. 2010, 2018. All Rights Reserved.
+# -*- coding: utf-8 -*-
+""" Alien Vault OTX module to Access OTX threat Intelligence Data"""
+import json
+import re
+import datetime
+import requests
+from requests.packages.urllib3.util import Retry
+from requests.adapters import HTTPAdapter
+
+TOTAL_RETRIES = 3
+BACK_OFF_FACTOR = 5
+
+class InvalidInputParam(Exception):
+    def __init__(self, value=None):
```

Comment:
Take the function inputs as a parameter here too. Check which of the inputs are None and adjust the error message accordingly:
```
def __init__(self, value=None, search_value, search_type, search_section):
```

Reply:
needs to discuss.

Reply:
ok